src/features/TodoFeatures/TodosIndex.js

```diff
@@ -0,0 +1,46 @@
+import React from "react";
+import { useDispatch, useSelector } from "react-redux";
+import "./Css/todoIndex.css";
+
+import { todosState } from "./todoSlice";
+import { toggleTodo } from "./todoSlice";
+import { visState } from "../VisibilityFeatures/visibilitySlice";
+
+const TodosIndex = () => {
+  const stateTodos = useSelector(todosState);
+  const dispatch = useDispatch();
+  const vis = useSelector(visState);
+  const filteredTodos = (arr) => {
+    if (vis === "Active") {
+      return arr.filter((todo) => !todo.completed);
+    }
+    if (vis === "Completed") {
+      return arr.filter((todo) => todo.completed);
+    } else {
+      return arr;
+    }
+  };
+
+  let todos = filteredTodos(stateTodos);
+  const clickHandler = (e) => {
+    dispatch(toggleTodo(e.target.id));
+  };
+
+  return (
+    <ul onClick={clickHandler}>
+      {todos.map((todo) => {
+        return todo.completed ? (
+          <li className="line" key={todo.id} id={todo.id}>
```

Comment:
To make this less repetitive you can add `className={todo.completed ? "line" : ""}` , adding that would provide the same result.